SEGV in `std::simd` tests

The CI often, but not deterministically, crashes inside a test using `std::simd`, with the following errors:
```
Randomness seeded to: 4119962326
AddressSanitizer:DEADLYSIGNAL

=================================================================
==8399==ERROR: AddressSanitizer: SEGV on unknown address 0x03e9000020cf (pc 0x55b54af5f43c bp 0x7ffd8de6d430 sp 0x7ffd8de6b780 T0)
==8399==The signal is caused by a READ memory access.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests is a Catch2 v3.4.0 host application.
Run with -? for options

-------------------------------------------------------------------------------
simd.Simd.stdsimd
-------------------------------------------------------------------------------
/home/runner/work/llama/llama/tests/simd.cpp:[13](https://github.com/alpaka-group/llama/actions/runs/6318373925/job/17170551919?pr=775#step:14:14)9
...............................................................................

/home/runner/work/llama/llama/tests/simd.cpp:[14](https://github.com/alpaka-group/llama/actions/runs/6318373925/job/17170551919?pr=775#step:14:15)1: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGSEGV - Segmentation violation signal

===============================================================================
test cases:   646 |   645 passed | 1 failed
assertions: 86551 | 86550 passed | 1 failed

AddressSanitizer:DEADLYSIGNAL
AddressSanitizer: nested bug in the same thread, aborting.
Error: Process completed with exit code 1.
```
This only happens on g++-12 and g++-13. I cannot reproduce it locally, though.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -170,8 +170,9 @@ jobs:
           - name: build-ubuntu-gcc10-nvcc11.6
             cxx: g++-10
             cuda_url: https://developer.download.nvidia.com/compute/cuda/11.6.2/local_installers/cuda_11.6.2_510.47.03_linux.run
-          - name: build-ubuntu-gcc11
+          - name: build-ubuntu-gcc11-asan
             cxx: g++-11
+            asan: ON
           - name: build-ubuntu-gcc11-nvcc11.7
             cxx: g++-11
             cuda_url: https://developer.download.nvidia.com/compute/cuda/11.7.1/local_installers/cuda_11.7.1_515.65.01_linux.run

--- a/tests/simd.cpp
+++ b/tests/simd.cpp
@@ -16,6 +16,8 @@
 #    warning "LLAMA SIMD tests disabled for libc++. Their std::experimental::simd implementation is incomplete"
 #elif !__has_include(<experimental/simd>)
 #    warning "LLAMA SIMD tests disabled. Need std::experimental::simd, which is available in libstdc++ since GCC 11"
+#elif defined(__GNUG__) && __GNUC__ >= 12 && defined(__SANITIZE_ADDRESS__) && defined(__AVX512F__)
+#    warning "With ASan enabled, g++-12 and later generates an aligned store to an unaligned address with AVX512"
 #else
 #    include <experimental/simd>
 

